### PR TITLE
[core] Modify GKE to generate code to be 

### PR DESCRIPTION
**Pull request by schweitzpgi**
used exclusively with argument conversion and argument synthesis.

This is the initial implementation to argument gathering on the C++ side. It changes the autogenerated C++ host entry point to gather the arguments that are passed in and pass them as a `std::vector<void*>` to the runtime for processing. The vector includes only textual arguments for now. All hidden arguments are ignored.

~#2040 must be committed first.~

### CODE REVIEW

File: lib/Optimizer/Transforms/GenKernelExecution.cpp [hunk 1]
```diff
@@ -1392,6 +1393,91 @@ class GenerateKernelExecution
...
+    Value loadKernName = builder.create<LLVM::AddressOfOp>(
+        loc, cudaq::opt::factory::getPointerType(kernelNameObj.getType()),
+        kernelNameObj.getSymName());
+    auto castKernelNameObj =
+        builder.create<cudaq::cc::CastOp>(loc, i8PtrTy, loadKernName);
+    builder.create<func::CallOp>(
+        loc, std::nullopt, cudaq::runtime::launchKernelVersion2FuncName,
+        ArrayRef<Value>{castKernelNameObj, castStdvec, castResultBuffer});
+
+    // FIXME: Drop any results on the floor for now and return random data left
```
Comment (annagrin) [inline lib/Optimizer/Transforms/GenKernelExecution.cpp:1469]: I thought we support returning values from kernels in c++ and python now?

Comment (schweitzpgi) [reply lib/Optimizer/Transforms/GenKernelExecution.cpp:1469]: I'll add this to the new flow in a later PR.

Comment (schweitzpgi) [reply lib/Optimizer/Transforms/GenKernelExecution.cpp:1469]: As to the question, it's nuanced.

Does the kernel launch have the _capability_ to handle return values? Yes. Is that capability fully functional and correct? Almost completely, if you consider C++. Do any of the kernel launchers deal with return values? Almost never. There is a singular path that can return a `int`, `double` and `bool` (only when remote simulation and other qualifiers IIRC).

**Review by 1tnguyen — APPROVED** [review]
LGTM 👍

**Pull request merged**